File: 2394-count-subarrays-with-score-less-than-k/solution.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) 
    {
        const int n = nums.size();
        long long windowSum = 0LL;
        long long numSubArrays = 0LL;
        
        int left = 0;
        for(int right = 0; right < n; right++)
        {
            windowSum += nums[right];
            while(left <= right && windowSum*(right-left+1) >= k)
            {
                windowSum -= nums[left++];
            }
            
            numSubArrays += (right-left+1);
            
        }
        
        return numSubArrays;
        
        
    }
};

File: 2394-count-subarrays-with-score-less-than-k/solution.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        windowSum = 0
        numSubArray = 0
        
        left      = 0
        for right, num in enumerate(nums):
            windowSum += num
            while left <= right and (windowSum)*(right-left+1) >= k:
                windowSum -= nums[left]
                left += 1

            length = right-left+1
            numSubArray += length

                        
        return numSubArray
        
